### PR TITLE
[#85933030] LogSearch

### PR DESCRIPTION
**Pull request by madkin10**

### CODE REVIEW

**Comment by madkin10** [comment]
#Log Search

- Added a "Log Search" class to search a log to find a string.

[Here is the card] (https://www.pivotaltracker.com/story/show/85933030)

File: lib/logSearch.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+
+class LogSearch
+
+  def search(log, stringToFind)
+
+    if not File.exist?(log)
```
Comment (jimcegelski) [inline lib/logSearch.rb]: You could consider changing `if not` to `unless`...which do you think is more readable

File: lib/logSearch.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      raise IOError,"The log \"#{log}\" does not exist."
+    end
+
+    print "The log \"#{log}\" was found."
```
Comment (jimcegelski) [inline lib/logSearch.rb]: I would remove this line...I don't think it adds value

File: spec/test.txt [hunk 3]
```diff
@@ -0,0 +1,28 @@
+2015-01-21 09:54:15.285|*** Server JUBILEE
```
Comment (jimcegelski) [inline spec/test.txt]: I would recommend this file be moved to the test files directory

File: lib/logSearch.rb (hunk 1 above)
Comment (vht-hfenton) [inline lib/logSearch.rb]: I agree with @jimcegelski , you should consider changing it to unless.

File: spec/logSearch_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+require 'rspec'
+require_relative '../lib/logSearch'
```
Comment (vht-hfenton) [inline spec/logSearch_spec.rb]: The spec_helper should be required here, the logSearch should be added to the spec_helper.

File: lib/logSearch.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+
+class LogSearch
```
Comment (vht-hfenton) [inline lib/logSearch.rb]: The class name is fine, but file names typically follow the convention 'log_search'.

File: spec/logSearch_spec.rb (hunk 4 above)
Comment (vht-hfenton) [inline spec/logSearch_spec.rb]: File names for specs also follow the snake case convention 'log_search_spec'.  (class_file_name_spec)

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -1,4 +1,6 @@
 require 'simplecov'
 SimpleCov.start
 
+$:.unshift File.dirname(__FILE__)+'../lib'
 
+require 'log_search'
```
Comment (vht-hfenton) [inline spec/spec_helper.rb]: @glyphrider do the require statements have to be before 'SimpleCov.start'?

**Pull request merged**